### PR TITLE
feat(locale): add hyphenated names to English locales

### LINKED ISSUE #691: feat(name): add hyphenated patterns to `lastName()`
### Clear and concise description of the problem

The ability to generate hyphenated last names so that it seems just a little bit more realistic. The function only generates single last names currently.

### Suggested solution

Add a boolean option ```hyphenated?```

### Alternative

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by matthewmayer**
fix #691
- default en locale gets 95% non-hyphenated 5% hyphenated
- other locales without existing last_name_patterns get 100% non-hyphenated
- locales with split male_last_name and  female_last_names get split patterns
- existing locales with last_name_patterns (e.g. lv, en_GB) are not changed.

If additional information about hyphenation rules in other locales can be found, this can be added in a subsequent PR. I kept this one formulaic to make it easy to review.

### CODE REVIEW

**Comment by matthewmayer** [comment]
this exposed a bug in the test for commitEntry, the local part of the email addresses was not allowing hyphens. i changed the regex to match the one we use in the emailAddress method https://github.com/faker-js/faker/blob/add2254ca31434af3003c0051064bdb27713992d/src/modules/internet/index.ts

**Comment by Shinigami92** [comment]
> this exposed a bug in the test for commitEntry, the local part of the email addresses was not allowing hyphens. i changed the regex to match the one we use in the emailAddress method [`add2254`/src/modules/internet/index.ts](https://github.com/faker-js/faker/blob/add2254ca31434af3003c0051064bdb27713992d/src/modules/internet/index.ts)

Please lets extract this to its own PR, so
1. it is reviewable in parallel to this PR
2. it gets its own "fix/test" changelog entry

**Comment by matthewmayer** [comment]
I considered that, but the test doesn't fail on the next branch because email addresses in en never contain hyphens.

**Comment by Shinigami92** [comment]
> I considered that, but the test doesn't fail on the next branch because email addresses in en never contain hyphens.

You are free to add static tests that handles special cases in the random seeded test cases (second part of each test file)

Maybe we need to change something here instead of the test
https://github.com/faker-js/faker/blob/11a5f51ddb4ab1163d770c4b12f5fe4711b8e11d/src/modules/git/index.ts#L123

We are already starting to pollute this PR with unrelated discussions, please lets stop this and extract it so we can have dedicated discussions

**Comment by matthewmayer** [comment]
> > I considered that, but the test doesn't fail on the next branch because email addresses in en never contain hyphens.
> 
> You are free to add static tests that handles special cases in the random seeded test cases (second part of each test file)
> 
> Maybe we need to change something here instead of the test
> 
> https://github.com/faker-js/faker/blob/11a5f51ddb4ab1163d770c4b12f5fe4711b8e11d/src/modules/git/index.ts#L123
> 
> We are already starting to pollute this PR with unrelated discussions, please lets stop this and extract it so we can have dedicated discussions

#1876

File: src/locales/lv/person/female_last_name_patterns.ts [hunk 1]
```diff
@@ -1,7 +1 @@
-export default [
-  { value: '{{person.female_last_name}}', weight: 8 },
-  {
-    value: '{{person.female_last_name}}-{{person.female_last_name}}',
-    weight: 2,
-  },
-];
```
Comment (ST-DDT) [inline src/locales/lv/person/female_last_name_patterns.ts]: Was this change intentional?

File: src/locales/lv/person/male_last_name_patterns.ts [hunk 2]
```diff
@@ -1,4 +1 @@
-export default [
-  { value: '{{person.male_last_name}}', weight: 8 },
-  { value: '{{person.male_last_name}}-{{person.male_last_name}}', weight: 2 },
-];
```
Comment (ST-DDT) [inline src/locales/lv/person/male_last_name_patterns.ts]: Same here

**Review by ST-DDT — COMMENTED** [review]
These lines are now somewhat mandatory:

https://github.com/faker-js/faker/blob/11a5f51ddb4ab1163d770c4b12f5fe4711b8e11d/src/modules/person/index.ts#L120-L135

So IMO the other code branch could be removed.
Also the PR does more than the PR title suggests IMO.

Not sure whether it is worth splitting into two PRs:
1. Make last_name_patterns required/Generate last_name_patterns for all locales
2. add hyphenated names to English locales

**Comment by matthewmayer** [comment]
It's not mandatory if you create a new locale which doesn't extend en.

File: src/locales/lv/person/male_last_name_patterns.ts (hunk 2 above)
Comment (matthewmayer) [reply src/locales/lv/person/male_last_name_patterns.ts]: reverted

File: src/locales/lv/person/female_last_name_patterns.ts (hunk 1 above)
Comment (matthewmayer) [reply src/locales/lv/person/female_last_name_patterns.ts]: reverted

**Comment by ST-DDT** [comment]
Please fix the merge conflicts.

**Comment by ST-DDT** [comment]
Please adjust the PR title to reflect the current. It also changes non English locales.

**Comment by matthewmayer** [comment]
> Please adjust the PR title to reflect the current. It also changes non English locales.

it doesn't add any hyphenated patterns to any non-English locales, the changes are to prevent hyphenated patterns from propagating to the other non-English locales unexpectedly.

**Review by ST-DDT — APPROVED** [review]

**Comment by matthewmayer** [comment]
could append ", add last_name_patterns to all locales" maybe? but makes it a bit long

**Review by Shinigami92 — APPROVED** [review]
